src/components/select/CustomSelect.tsx

```diff
@@ -0,0 +1,163 @@
+import classNames from "classnames";
+import { useEffect, useId, useLayoutEffect, useRef, useState } from "react";
+import type { FC, ReactNode } from "react";
+import {
+  ClassName,
+  Field,
+  ContextualMenu,
+  PropsWithSpread,
+  FieldProps,
+} from "@canonical/react-components";
+import CustomSelectDropdown, {
+  CustomSelectOption,
+  getOptionText,
+} from "./CustomSelectDropdown";
+import useEventListener from "@use-it/event-listener";
+import { adjustDropdownHeight } from "util/customSelect";
+
+export type Props = PropsWithSpread<
+  FieldProps,
+  {
+    // Selected option value
+    value: string;
+    // Array of options that the select can choose from.
+    options: CustomSelectOption[];
+    // Function to run when select value changes.
+    onChange: (value: string) => void;
+    // id for the select component
+    id?: string | null;
+    // Name for the select element
+    name?: string;
+    // Whether if the select is disabled
+    disabled?: boolean;
+    // The content for success validation.
+    success?: ReactNode;
+    // The content for error validation.
+    error?: ReactNode;
+    // Help text to show below the field.
+    help?: ReactNode;
+    // Styling for the wrapping Field component
+    wrapperClassName?: ClassName;
+    // The styling for the select toggle button
+    toggleClassName?: ClassName;
+    // Whether the select is searchable. Option "auto" is the default, the select will be searchable if it has 5 or more options.
+    searchable?: "auto" | "always" | "never";
+    // Whether to focus on the element on initial render.
+    takeFocus?: boolean;
+  }
+>;
+
+const CustomSelect: FC<Props> = ({
+  value,
+  options,
+  onChange,
+  id,
+  name,
+  disabled,
+  success,
+  error,
+  help,
+  wrapperClassName,
+  toggleClassName,
+  searchable = "auto",
+  takeFocus,
+  ...fieldProps
+}) => {
+  const [isOpen, setIsOpen] = useState(false);
+  const validationId = useId();
+  const defaultSelectId = useId();
+  const selectId = id || defaultSelectId;
+  const helpId = useId();
+  const hasError = !!error;
+  const searchRef = useRef<HTMLInputElement>(null);
+  const dropdownRef = useRef<HTMLUListElement>(null);
+
+  useEffect(() => {
+    if (takeFocus) {
+      const toggleButton = document.getElementById(selectId);
+      toggleButton?.focus();
+    }
+  }, [takeFocus]);
```

Comment:
It works, when use the keyboard navigation, but with mouse the blue outline is missing.

Reply:
thanks! Looking into it now

Reply:
@edlerd the above issue is resolved, in addition I also noticed the following issues which are now fixed:
1. The edit permission panel was always focusing on the entitlement selector on first render. This was a pre-existing issue actually, now we will always focus on the resource type selector on first render.
2. When a select dropdown is open, keyboard event was not triggering for a dropdown without the search input. This is because the custom select uses a `div` for the dropdown container, which is not by default focusable and therefore cannot capture keyboard events. This is resolved by adding tabindex to the `div` container and focusing on it if the select is not searchable.